src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableSqlConnection.cs

```diff
@@ -123,6 +125,21 @@ private bool IsSqlDwConnection(IDbConnection conn)
             return _isSqlDwDatabase;
         }
 
+        /// <summary>
+        /// Determines if a connection is being made to SQLOnDemand.
+        /// </summary>
+        /// <param name="conn">A connection object.</param>
+        private bool IsSqlOnDemandConnection(IDbConnection conn)
```

Comment:
>IsSqlOnDemandConnection [](start = 21, length = 23)

Now we're running the exact same query twice (once for SQL DW and once for On Demand). It'd be better to just query once and store the engine edition information for both after making that call.

Reply:
I think everything is generalized properly now.